### ISSUE TITLE
gh-pages deployment action failing

### ISSUE THREAD

**Issue by avats-dev** [BitByte-TPC/first-bit#30]
Since some time (to be exact, since last two merges), automatic deployment of gh-pages is failing.

The error from logs look like this:
```
yarn run v1.22.5
$ vue-cli-service build
-  Building for production...

 ERROR  Failed to compile with 1 errors12:31:11 PM

This relative module was not found:

* ../../public/nameDB.json in ./node_modules/cache-loader/dist/cjs.js??ref--12-0!./node_modules/thread-loader/dist/cjs.js!./node_modules/babel-loader/lib!./node_modules/cache-loader/dist/cjs.js??ref--0-0!./node_modules/vue-loader/lib??vue-loader-options!./src/components/Grid.vue?vue&type=script&lang=js&
 ERROR  Build failed with errors.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
##[error]Process completed with exit code 1.
```

### Possible reasons:
- outdated package or any compatibility issues
- not considering case sensitive package names

### Possible Resolution
I don't know much but what I found is that following methods might solve this (not sure):
Reference : https://github.com/vuejs/vue-cli/issues/439

```
rm -rf node_modules
npm install
```

or

```
npm i -g npm-check-updates
npm-check-updates -u
npm install
```

These might help resolve package issues if there would be any.
Also @Janglee123 why are we not using npm instead of yarn?

**Comment by Janglee123** [comment]
The error says `../../public/nameDB.json` was not found. Which simply means there is no `nameDB.json`. The step that compiles all name.json files into the one single nameDB.json file is not working and throwing an error
```
(node:2723) UnhandledPromiseRejectionWarning: Error: 404 error making request get api.github.com/users/Rishabh Devbanshi: "Not Found"
```
The root of the problem is here https://github.com/BitByte-TPC/first-bit/blob/009200fb296b919530ae9733beedb73a5b238df4/public/directory/RishabhDevbanshi.json#L2
This is not a valid GitHub ID. Seems like we also have to check `githubId` field before merging a pr.

>  Also @Janglee123 why are we not using npm instead of yarn?

Yarn executes tasks in parallel and much faster than npm especially while installing.

**Comment by Janglee123** [comment]
@avats-dev Add what is a correct GitHub ID in the guide. Additionally, I found many are not changing `bio` field and pushes changes with the default line, so also add it in the guide to write a small bio, or pr won't be get merged now onwards.

**Comment by garg3133** [comment]
Seems like wrong bio in `shivamrai2001.json` is causing problem in later checks also.
![shivamrai2001](https://user-images.githubusercontent.com/39924567/94155852-0a0a7700-fe9d-11ea-8a4f-748a3dda2e20.png)

Changing his bio myself.

**Comment by avats-dev** [comment]
@Janglee123 yeah, I'll add a reference as to what a GithubId is in the guide. Also thanks for the tip on yarn.

Apart from this, in  https://github.com/BitByte-TPC/first-bit/pull/32 , first his checks were failing due to wrong bio. Then he edited someone else's bio, and then checks passed, how?

Is it like if any of the previous PRs are invalid, next ones will get stuck too?

Also, @garg3133 :
> Seems like wrong bio in `shivamrai2001.json` is causing problem in later checks also.
> ![shivamrai2001](https://user-images.githubusercontent.com/39924567/94155852-0a0a7700-fe9d-11ea-8a4f-748a3dda2e20.png)
> 
> Changing his bio myself.

What's wrong in his bio? I'm not getting this.

**Comment by garg3133** [comment]
> Is it like if any of the previous PRs are invalid, next ones will get stuck too?

Yes, GitHub actions is checking all the already existing json files also in each PR.

> What's wrong in his bio? I'm not getting this.

Actually, the checks in #31 had already failed due to this file before its bio was fixed in #32. So, it had already been fixed when I edited it and there was nothing wrong in this file. 

But as I didn't read the changes in GitHub actions, I thought the bio needs to be completely changed. So, I changed it.
